packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,139 @@
+const spacingUnit = '--DataGrid-t-spacing-unit';
+
+// NOTE: Breakpoints can't come from the theme because we need access to them at
+// initialization time and media-queries can't use CSS variables. For users with
+// custom breakpoints, we might want to provide a way to configure them globally
+// instead of through the theme.
+const breakpoints = {
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+export const vars = {
+  /* Helpers */
+  /** @usage `...vars.props(vars.typography.body)` */
+  props,
+  breakpoints,
+  spacing,
+  spacingUnit,
+
+  /* Variables */
+  colors: {
+    border: {
+      base: '--DataGrid-t-colors-border-base',
+    },
+    foreground: {
+      base: '--DataGrid-t-colors-foreground-base',
+      muted: '--DataGrid-t-colors-foreground-muted',
+      accent: '--DataGrid-t-colors-foreground-accent',
+      disabled: '--DataGrid-t-colors-foreground-disabled',
+    },
+    background: {
+      base: '--DataGrid-t-colors-background-base',
+      // FIXME(kenan): validate `pinned`
+      pinned: '--DataGrid-t-colors-background-pinned',
```

Comment:
@KenanYusuf I've added this.

Reply:
For element specific things like this, I would add a new key for the element e.g. `cell`, and nest variables in there:
```ts
colors: {
},
cell: {
  background: {
    pinned: "--DataGrid-t-cell-background-pinned"
  }
}
```

T